src/battle_util.c

```diff
@@ -2084,6 +2084,24 @@ void TryToRevertMimicry(void)
     }
 }
 
+u32 GetMonFriendshipScore(struct Pokemon *pokemon) // Based on GetLeadMonFriendshipScore
```

Comment:
```suggestion
u32 GetMonFriendshipScore(struct Pokemon *pokemon)
```
I would modify `GetLeadMonFriendshipScore` to use `GetMonFriendshipScore` to keep consistent.

Reply:
> I would modify `GetLeadMonFriendshipScore` to use `GetMonFriendshipScore` to keep consistent.

Hmm... how so? 🤔 

The functions are practically the same. The only difference is that my copy allows to freely load a Pokémon in the party instead of force checking the very first one.

Reply:
Because copying functions instead of using already existing ones steers the code into becoming unmantainable. You shouldn't have to go to multiple functions and modify them all if they're all supposed do the same thing.

Reply:
> Because copying functions instead of using already existing ones steers the code into becoming unmantainable. You shouldn't have to go to multiple functions and modify them all if they're all supposed do the same thing.

I was rather asking how to go about it 😅 

One function comes from vanilla and specifically checks the lead Pokémon in the party because it's called in overworld scripts as a `special`, and those cannot take any extra parameters.

I'm not sure how should I go about "merging the functions" without also having to modify every overworld script in which the `special GetLeadMonFriendshipScore` is called.

Reply:
```c
u8 GetLeadMonFriendshipScore(void)
{
    struct Pokemon *pokemon = &gPlayerParty[GetLeadMonIndex()];
    return GetMonFriendshipScore(pokemon);
}
```